mysql.py

```diff
@@ -36,3 +36,36 @@ def fix_replication_from_slow_query_log_after_upgrade():
     run_mysql_command("START SLAVE;")
     run_mysql_command("SET GLOBAL slow_query_log = 'ON';")
     run_mysql_command("show slave status\G;")
+
+
+@task
+def setup_slave_from_master(master):
+    """
+    Sets up a slave from a master by taking a dump from the master,
+    copying it to the slave and then restoring the dump.
+
+    Usage: fab environment -H mysql-slave-1.backend mysql.setup_slave_from_master:'mysql-master-1.backend'
+    """
+    if len(env.hosts) > 1:
+        print 'This job is currently only setup to run against one slave at a time'
+
+    with settings(host_string=master):
+        # The use of `--master-data` here implies `--lock-all-tables` per the
+        # MySQL reference manual: http://dev.mysql.com/doc/refman/5.1/en/mysqldump.html#option_mysqldump_master-data
+        run('sudo -i mysqldump -u root --all-databases --master-data --add-drop-database > dump.sql')
+
+    with settings(host_string=master, forward_agent=True):
+        run('scp dump.sql {0}:~'.format(env.hosts[0]))
+
+    with settings(host_string=master):
+        run('rm dump.sql')
```

Comment:
We can see this is ok by creating a task like this and checking the permissions of the file:

```python
@task
def test_perms():
    run('sudo -i echo test > test-perms')
    run('ls -l')
    run('rm test-perms')
```

Which gives this output:

```
[HOST] 08:39:11 ~/govuk/fabric-scripts (add-mysql-replication-setup-task) $ fab preview -H frontend-1.frontend mysql.test_perms
[frontend-1.frontend] Executing task 'preview'
Disconnecting from jumpbox.preview.alphagov.co.uk... done.
[frontend-1.frontend] Executing task 'mysql.test_perms'
[frontend-1.frontend] run: sudo -i echo test > test-perms
[frontend-1.frontend] run: ls -l
[frontend-1.frontend] out: total 4
[frontend-1.frontend] out: -rw-r--r-- 1 alexmuller alexmuller 5 Sep  3 07:39 test-perms
[frontend-1.frontend] out: 

[frontend-1.frontend] run: rm test-perms
Disconnecting from jumpbox.preview.alphagov.co.uk... done.
Disconnecting from frontend-1.frontend... done.

Done.
```